test/controllers/CalculationControllerSpec.scala

```diff
@@ -203,11 +203,26 @@ class CalculationControllerSpec extends TestSupport with MockCalculationService
             enable(TaxYearOverviewUpdate)
             mockSingleBusinessIncomeSource()
             mockCalculationNotFound()
+            mockFinancialDetailsSuccess()
+            mockGetReportDeadlines(fromDate = LocalDate.of(testYear - 1, 4, 6),
+              toDate = LocalDate.of(testYear, 4, 5))(
+              response = testObligtionsModel
+            )
+
+            val expectedContent: String = views.html.taxYearOverview(
+              testYear,
+              None,
+              testChargesList,
+              testObligtionsModel,
+              mockImplicitDateFormatter,
+              taxYearsBackLink).toString
 
             val result = TestCalculationController.renderTaxYearOverviewPage(testYear)(fakeRequestWithActiveSession)
 
-            status(result) shouldBe Status.INTERNAL_SERVER_ERROR
+            status(result) shouldBe Status.OK
+            contentAsString(result) shouldBe expectedContent
             contentType(result) shouldBe Some("text/html")
+            result.session.get(SessionKeys.chargeSummaryBackPage) shouldBe Some("taxYearOverview")
           }
```

Comment:
Optional suggestion: please change the description of the test case to a more relevant now. Currently it says `"return the internal server error page"`, but it's not what is verified in this test case.

Reply:
Done